### PR TITLE
Activate TestTransportManagerDefault_Init_LastStateUsed test

### PR DESCRIPTION
**Pull request by vlantonov**
The reported deadlock on thread create under Windows Qt was not reproduced.

Related issue: [APPLINK-27939](https://adc.luxoft.com/jira/browse/APPLINK-27939)

### CODE REVIEW

**Comment by vlantonov** [comment]
Please review: @AByzhynar, @OHerasym, @VProdanov, @wolfylambova22, @pvvasilev

**Comment by vlantonov** [comment]
@AByzhynar, @OHerasym, @VProdanov, @wolfylambova22, @pvvasilev please pay attention that only the last commit should be reviewed as previous ones are part of the pull request already reviewed by you - namely #820 and #822 .

File: src/components/application_manager/include/application_manager/policies/policy_retry_sequence.h [hunk 1]
```diff
@@ -42,7 +42,8 @@ class PolicyHandler;
 class RetrySequence : public threads::ThreadDelegate {
  public:
   explicit RetrySequence(PolicyHandler* const policy_handler);
-  void threadMain();
+  void threadMain() OVERRIDE;
+  void exitThreadMain() OVERRIDE;
 
  private:
   PolicyHandler* const policy_handler_;
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/policies/policy_retry_sequence.h]: @vlantonov Why do we need const pointer?

File: src/components/hmi_message_handler/src/mqueue_adapter.cc [hunk 2]
```diff
@@ -48,14 +49,22 @@ class ReceiverThreadDelegate : public threads::ThreadDelegate {
...
  private:
-  virtual void threadMain() {
-    while (true) {
+  void threadMain() OVERRIDE {
+    while (!shutdown_requested_) {
       static char buffer[kMqueueMessageSize];
-      const ssize_t size =
-          mq_receive(mqueue_descriptor_, buffer, kMqueueMessageSize, NULL);
+
+      struct timespec current_time;
```
Comment (AByzhynar) [inline src/components/hmi_message_handler/src/mqueue_adapter.cc]: @vlantonov The `<time.h>` header declares the structure timespec and functions below. And looks that it's missed

File: src/components/application_manager/include/application_manager/policies/policy_retry_sequence.h (hunk 1 above)
Comment (vlantonov) [inline src/components/application_manager/include/application_manager/policies/policy_retry_sequence.h]: @AByzhynar I do not know. This is just an interface without implementation created by @justinjdickow I decided just to do some fixes in respect to proper thread stopping.

File: src/components/hmi_message_handler/src/mqueue_adapter.cc (hunk 2 above)
Comment (vlantonov) [inline src/components/hmi_message_handler/src/mqueue_adapter.cc]: @AByzhynar Fixed, see 4b1f7284ea5aa9d01880f49951562a83021bd254

**Comment by AByzhynar** [comment]
@vlantonov Please fix merge conflicts

**Comment by vlantonov** [comment]
@AByzhynar Done fixing merge conflicts.

**Pull request merged**